### PR TITLE
Implement auto installation of core and enterprise plugins

### PR DESCRIPTION
**Pull request by bamarni**
https://jira.mesosphere.com/browse/DCOS_OSS-4694

Example: https://asciinema.org/a/OBcbzhK7DEKBW0SKxNRo2JBxt

### CODE REVIEW

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

File: pkg/cmd/dcos.go [hunk 1]
```diff
@@ -158,48 +144,10 @@ func newPluginCommand(ctx api.Context, cmd plugin.Command, isDummyCoreCommand bo
...
-		},
-	}
-
-}
-
 // extractCorePlugin extracts the bundled core plugin into the plugins folder.
 func extractCorePlugin(ctx api.Context, cluster *config.Cluster) (*plugin.Plugin, error) {
-	ctx.Logger().Warn(`Extracting "dcos-core-cli"...`)
+	ctx.Logger().Warn(`Extracting "dcos-core-cli"...
+This setup is deprecated, see https://docs.mesosphere.com/1.12/cli/experiments/#automatic-installation-of-core-and-enterprise-cli-plugins for more information.`)
```
Comment (bamarni) [inline pkg/cmd/dcos.go]: I put this URL for now but we should have a new section for 1.13 as it's not an experiment anymore.

**Comment by ArmandGrillet** [comment]
Failing Windows CI:
```
STDERR: 
---------------------------- Captured stdout call -----------------------------
CMD: ['dcos', 'plugin', 'add', 'C:\\windows\\workspace\\tests\\integration\\fixtures\\plugins\\dcos-test\\win32\\dcos-test.exe']
STDOUT: 
STDERR: 
CMD: ['dcos', 'plugin', 'remove', 'dcos-test']
STDOUT: 
STDERR: Error: remove C:\Users\jenkins\.dcos\clusters\1ec83651-26f6-4b04-be0d-d38fe53a6bd1\subcommands\dcos-test\env\bin\dcos-test.exe: Access is denied.

-------------------------- Captured stdout teardown ---------------------------
CMD: ['dcos', 'cluster', 'remove', '1ec83651-26f6-4b04-be0d-d38fe53a6bd1']
STDOUT: 
STDERR: 
==================== 1 failed, 16 passed in 78.08 seconds =====================
```

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

File: pkg/cmd/dcos.go [hunk 2]
```diff
@@ -32,10 +33,33 @@ const annotationUsageOptions string = "usage_options"
...
+		Args: cobra.ArbitraryArgs,
 		PersistentPreRun: func(cmd *cobra.Command, args []string) {
 			cmd.SilenceUsage = true
 		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return cmd.Help()
+			}
+			switch args[0] {
+			case "job", "marathon", "node", "package", "service", "task":
```
Comment (rgo3) [inline pkg/cmd/dcos.go]: Is there a reason why we don't extract the plugin during the setup and remove the overhead when running those commands the first time?

Comment (bamarni) [reply pkg/cmd/dcos.go]: The plugin is installed during setup, this is only for the scenario when a user updates their CLI and they have a previously configured cluster (without the core-cli plugin).

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

File: pkg/setup/setup.go [hunk 3]
```diff
@@ -293,38 +293,89 @@ func (s *Setup) installDefaultPlugins(httpClient *httpclient.Client) error {
...
-	// The installation of the core and EE plugins happen in parallel.
-	// We wait for the installation of the enterprise plugin before returning.
-	errEnterprise := <-enterpriseInstallErr
-
+	errCore := s.installPlugin("dcos-core-cli", httpClient, version, pbar)
+	wg.Done()
+	pbar.Wait()
 	if errCore != nil {
-		return errCore
+		errCore = corecli.InstallPlugin(s.fs, s.pluginManager)
```
Comment (ArmandGrillet) [inline pkg/setup/setup.go]: Why are we setting `errCore` again if it's not `nil`?

Comment (bamarni) [reply pkg/setup/setup.go]: I've added a comment, it is about extracting the dcos-core-cli when the installation through S3 / Cosmos failed.

**Comment by bamarni** [comment]
@mesosphere-ci run integration tests

**Pull request merged**